tasks/main.yml

```diff
@@ -23,14 +23,27 @@
 
 # The user is created by installing pacemaker packages. We just need to set the
 # password.
-- name: Set hacluster password
-  user:
-    name: hacluster
-    password: "{{
-      ha_cluster_hacluster_password | string |
-      password_hash('sha512', ansible_hostname.replace('-', 'x')[:16]) }}"
+- name: Provide a password for the hacluster user
   when:
     - ha_cluster_hacluster_password | string | length > 0
+  block:
+    - name: Generate a password hash
+      # The arg `-6` means SHA512 based algorithms.
+      command:
+        cmd: >-
+          openssl passwd
+            -6
+            -salt {{ ansible_hostname.replace('-', 'x') }}
+            {{ ha_cluster_hacluster_password | string }}
+      register: __ha_cluster_openssl_call_result
+      changed_when: false
+      failed_when: __ha_cluster_openssl_call_result.rc != 0
```

Comment:
I think this is the default Ansible behavior, so need to state this `failed_when`.

Reply:
Thanks, I removed the line.